crates/iceberg/src/expr/visitors/expression_evaluator.rs

```diff
@@ -273,7 +273,7 @@ mod tests {
             ))])
             .build()?;
 
-        let spec = BoundPartitionSpec::builder(schema.clone())
+        let spec = PartitionSpec::builder(schema.clone())
             .with_spec_id(1)
             .add_unbound_fields(vec![UnboundPartitionField::builder()
```

Comment:
Bikeshedding alert! I noticed this one when working on the code as well:
```suggestion
            .add_unbound_field(UnboundPartitionField::builder()
```